src/entities/Client.ts

```diff
@@ -0,0 +1,38 @@
+import { Entity, Column, BeforeInsert, BeforeUpdate, CreateDateColumn, UpdateDateColumn, JoinColumn, ManyToOne, PrimaryColumn, BaseEntity} from "typeorm";
+import * as bcrypt from "bcrypt";
+import { Role } from "./Role";
+@Entity('users')
+export class Client extends BaseEntity {
+    @PrimaryColumn({
+        unique: true
+    })
+    user_id!: number;
```

Comment:
this should be name 'id' and its type to be a uuid . if the user id is incremental and easy to be guessed there might be revealing of other users data without auth or authentication. for example if you have an api that takes the user id to get user's info, someone may attack this api to get all of our users info accessing private data.

Reply:
google that typeorm can create the id for u and how to customize that

Reply:
Oh okayy, I will check it out.